### PR TITLE
Fix path to freetype library in SharpFont's .dll mapping for OS X

### LINKED ISSUE #7911: Fails to load - System.TypeInitializationException - Mac OS X 10.10
I get the following when trying to run the game. I tried at first running the .app version but that was failing. So I did some digging and found I could command line it using mono from the resources folder. This is the result with the debug flag turned on. There is no X11 folder within /usr on Mac OS X 10.10. 

$ mono --debug OpenRA.Game.exe Game.Mod=ra
Platform is OSX
Using SDL 2 with OpenGL renderer
Desktop resolution: 1440x900
No custom resolution provided, using desktop resolution
Using resolution: 1440x900
Detected OpenGL version: 2.1
Using OpenAL sound engine
Using default device
Available mods:
	cnc: Tiberian Dawn (playtest-20150408)
	d2k: Dune 2000 (playtest-20150408)
	modchooser: Mod Chooser (playtest-20150408)
	ra: Red Alert (playtest-20150408)
Loading mod: ra
Exception of type `System.TypeInitializationException`: An exception was thrown by the type initializer for OpenRA.Graphics.SpriteFont
Inner Exception of type `System.DllNotFoundException`: /usr/X11/lib/libfreetype.6.dylib
  TypeName=``
    at (wrapper managed-to-native) SharpFont.FT:FT_Init_FreeType (intptr&)
  at SharpFont.Library..ctor () [0x00000] in <filename unknown>:0 
  at OpenRA.Graphics.SpriteFont..cctor () [0x00000] in <filename unknown>:0 
  at OpenRA.Graphics.Renderer.<InitializeFonts>m__1 (KeyValuePair`2 x) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.ToDictionary[KeyValuePair`2,String,SpriteFont] (IEnumerable`1 source, System.Func`2 keySelector, System.Func`2 elementSelector, IEqualityComparer`1 comparer) [0x0002f] in /private/tmp/source-mono-mac-3.12.99-branch-32/bockbuild-mono-3.12.0-branch/profiles/mono-mac-xamarin/build-root/mono-3.12.1/mcs/class/System.Core/System.Linq/Enumerable.cs:2928 
  at System.Linq.Enumerable.ToDictionary[KeyValuePair`2,String,SpriteFont] (IEnumerable`1 source, System.Func`2 keySelector, System.Func`2 elementSelector) [0x00000] in /private/tmp/source-mono-mac-3.12.99-branch-32/bockbuild-mono-3.12.0-branch/profiles/mono-mac-xamarin/build-root/mono-3.12.1/mcs/class/System.Core/System.Linq/Enumerable.cs:2915 
  at OpenRA.Graphics.Renderer.InitializeFonts (OpenRA.Manifest m) [0x00000] in <filename unknown>:0 
  at OpenRA.Game.InitializeMod (System.String mod, OpenRA.Arguments args) [0x00000] in <filename unknown>:0 
  at OpenRA.Game.Initialize (OpenRA.Arguments args) [0x00000] in <filename unknown>:0 
  at OpenRA.Program.Run (System.String[] args) [0x00000] in <filename unknown>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
Exception of type `System.DllNotFoundException`: /usr/X11/lib/libfreetype.6.dylib
TypeName=``
  at (wrapper managed-to-native) SharpFont.FT:FT_Done_FreeType (intptr)
  at SharpFont.Library.Dispose (Boolean disposing) [0x00000] in <filename unknown>:0 
  at SharpFont.Library.Finalize () [0x00000] in <filename unknown>:0 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/10831705-fails-to-load-system-typeinitializationexception-mac-os-x-10-10?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by obrakmann**
Fixes #7911

### CODE REVIEW

**Comment by pchote** [comment]
It might also be worth seeing if @Robmaister wants to make the same change upstream.  That won't help us in the short term, however, as SharpFont 3.0.0 doesn't work under OSX (#7696).

File: thirdparty/fetch-thirdparty-deps.sh [hunk 1]
```diff
@@ -42,6 +42,7 @@ if [ ! -f SharpFont.dll ]; then
 	nuget install SharpFont -Version 2.5.0.1
 	cp ./SharpFont.2.5.0.1/lib/net20/SharpFont* .
 	cp ./SharpFont.2.5.0.1/Content/SharpFont.dll.config .
+	sed -i '/osx/s@\(dll="\)[^"]*\(" />\)@\1/Library/Frameworks/Mono.framework/Libraries/libfreetype.6.dylib\2@' SharpFont.dll.config
```
Comment (pchote) [inline thirdparty/fetch-thirdparty-deps.sh]: This usage of `-i` is a nonstandard GNUism, so it doesn't work with OSX's BSD sed.

**Comment by obrakmann** [comment]
Right, forgot about that. Changed the sed command to do its thing like we did in the olden days ;)

**Comment by Robmaister** [comment]
@pchote Yup, I'll go ahead and make that change upstream and publish an updated version to NuGet, probably sometime tomorrow or on Monday.

**Comment by Mailaender** [comment]
I submitted it myself. https://github.com/Robmaister/SharpFont/issues/52 We should have done that earlier.

**Comment by pchote** [comment]
:+1: confirmed that this writes a correct path into SharpFont.dll.config

**Comment by Mailaender** [comment]
Doesn't break things on Linux. :white_check_mark:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/f9c7286970aaa4c6f45bcf22a298b0a965118b6d...4cfb05793a9667326a78284e56ffaf1fde35db0d)

**Pull request merged**